crates/primitives/src/lib.rs

```diff
@@ -95,6 +96,30 @@ pub fn sphere_data(assets: &AssetCache, sphere: &UVSphereMesh) -> Entity {
         .with(world_bounding_sphere(), bound_sphere)
 }
 
+
+pub fn torus_data(assets: &AssetCache, torus: &TorusMesh) -> Entity {
+    let aabb = AABB {
+        min: vec3(-torus.inner_radius, -torus.inner_radius, -torus.outer_radius),
+        max: vec3(torus.inner_radius, torus.inner_radius, torus.outer_radius),
+    };
```

Comment:
I believe this AABB is too small for a torus. X and Y bounds should be `+/-(torus.inner_radius + torus.outer_radius)`.

Reply:
i've fixed so that :